kotlin/display-ogc-api-collection/src/main/java/com/esri/arcgisruntime/sample/displayogcapicollection/MainActivity.kt

```diff
@@ -0,0 +1,155 @@
+        // in this mode, the table must be manually populated - panning and zooming won't request features automatically
+        ogcFeatureCollectionTable.featureRequestMode =
+            ServiceFeatureTable.FeatureRequestMode.MANUAL_CACHE
+
+        // load the table
+        ogcFeatureCollectionTable.loadAsync()
+
+        // ensure the feature collection table has loaded successfully before creating a feature layer from it to display on the map
+        ogcFeatureCollectionTable.addDoneLoadingListener {
+
+            if (ogcFeatureCollectionTable.loadStatus == LoadStatus.LOADED) {
+
+                // create a feature layer and set a renderer to it to visualize the OGC API features
+                val featureLayer = FeatureLayer(ogcFeatureCollectionTable)
+                val simpleRenderer =
+                    SimpleRenderer(SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, Color.BLUE, 3F))
+                featureLayer.renderer = simpleRenderer
+
+                // add the layer to the map
+                map.operationalLayers.add(featureLayer)
+
+                // zoom to a small area within the dataset by default
+                val datasetExtent = ogcFeatureCollectionTable.extent
+                if (datasetExtent != null && !datasetExtent.isEmpty) {
+                    mapView.setViewpointGeometryAsync(
+                        Envelope(
+                            datasetExtent.center,
+                            datasetExtent.width / 3,
+                            datasetExtent.height / 3
+                        )
+                    )
+                }
+            } else {
+                // show an alert if there is a loading failure
+                Log.e(
+                    ogcFeatureCollectionTable.loadError.message,
+                    ogcFeatureCollectionTable.loadError.additionalMessage
+                )
+                Toast.makeText(
+                    applicationContext,
+                    "Failed to load OGC Feature Collection Table",
+                    Toast.LENGTH_SHORT
+                ).show()
+            }
+        }
+
+        // once the map view navigation has completed, query the OGC API feature table for
+        // additional features within the new visible extent
+        mapView.addNavigationChangedListener {
+            if (!it.isNavigating) {
+
+                // get the current extent
+                val currentExtent = mapView.visibleArea.extent
+
+                // create a query based on the current visible extent
+                val visibleExtentQuery = QueryParameters()
+                visibleExtentQuery.geometry = currentExtent
+                visibleExtentQuery.spatialRelationship =
+                    QueryParameters.SpatialRelationship.INTERSECTS
+                // set a limit of 5000 on the number of returned features per request, the default on some services
+                // could be as low as 10
+                visibleExtentQuery.maxFeatures = 5000
```

Comment:
```suggestion
                val visibleExtentQuery = QueryParameters().apply {
               geometry = currentExtent
                spatialRelationship =
                    QueryParameters.SpatialRelationship.INTERSECTS
                // set a limit of 5000 on the number of returned features per request, the default on some services
                // could be as low as 10
                maxFeatures = 5000
                }
```

Just spotted a good opportunity to use a scoping function here :)